src/org/javarosa/xform/parse/StandardBindAttributesProcessor.java

```diff
@@ -48,39 +54,21 @@ DataBinding createBinding(IXFormParserFunctions parserFunctions, FormDef formDef
 
         binding.setDataType(getDataType(element.getAttributeValue(null, "type")));
 
-        final String xpathRel = element.getAttributeValue(null, "relevant");
-        if (xpathRel != null) {
-            if ("true()".equals(xpathRel)) {
-                binding.relevantAbsolute = true;
-            } else if ("false()".equals(xpathRel)) {
-                binding.relevantAbsolute = false;
-            } else {
-                binding.relevancyCondition = getCondition(formDef, ref, xpathRel, "relevant");
-            }
-        }
-
-        final String xpathReq = element.getAttributeValue(null, "required");
-        if (xpathReq != null) {
-            if ("true()".equals(xpathReq)) {
-                binding.requiredAbsolute = true;
-            } else if ("false()".equals(xpathReq)) {
-                binding.requiredAbsolute = false;
-            } else {
-                binding.requiredCondition = getCondition(formDef, ref, xpathReq, "required");
-            }
-        }
-
-        final String xpathRO = element.getAttributeValue(null, "readonly");
-        if (xpathRO != null) {
-            if ("true()".equals(xpathRO)) {
-                binding.readonlyAbsolute = true;
-            } else if ("false()".equals(xpathRO)) {
-                binding.readonlyAbsolute = false;
-            } else {
-                binding.readonlyCondition = getCondition(formDef, ref, xpathRO, "readonly");
-            }
-        }
-
+        setBoolAndCondition(ref, element, "relevant", formDef, new BoolAndConditionSetter() {
```

Comment:
Makes sense, it does look like "code that is idiomatic in another language/context". I think if there were 17 or maybe even 7 or those things I'd be convinced that an abstraction was necessary but in this case it feels more like an obfuscation to me. :/